server/src/main/java/org/apache/druid/discovery/BaseNodeRoleWatcher.java

```diff
@@ -76,28 +76,24 @@ public BaseNodeRoleWatcher(
       NodeRole nodeRole
   )
   {
-    this(listenerExecutor, nodeRole, DEFAULT_TIMEOUT_SECONDS);
+    this.nodeRole = nodeRole;
+    this.listenerExecutor = listenerExecutor;
   }
 
-  BaseNodeRoleWatcher(
+  public static BaseNodeRoleWatcher create(
       ScheduledExecutorService listenerExecutor,
-      NodeRole nodeRole,
-      long timeout
+      NodeRole nodeRole
   )
   {
-    this.nodeRole = nodeRole;
-    this.listenerExecutor = listenerExecutor;
-    this.listenerExecutor.schedule(
-        this::cacheInitializedTimedOut,
-        timeout,
-        TimeUnit.SECONDS
-    );
+    BaseNodeRoleWatcher nodeRoleWatcher = new BaseNodeRoleWatcher(listenerExecutor, nodeRole);
+    scheduleTimeout(nodeRoleWatcher, listenerExecutor, DEFAULT_TIMEOUT_SECONDS);
```

Comment:
See other comment about making `scheduleTimeout` an instance method; then this would simply be `nodeRoleWatcher.scheduleTimeout(DEFAULT_TIMEOUT_SECONDS);`

Reply:
Done